<launch>
	<arg name="output" default="log"/>
	<arg name="publish_clock" default="false"/>
	<arg name="teleop" default="true"/>
	<arg name="host" default="localhost" />

	<node name="airsim_node" pkg="airsim_ros_pkgs" type="airsim_node" output="$(arg output)">
		<param name="is_vulkan" type="bool" value="true" /> 
		<!-- ROS timer rates. Note that timer callback will be processed at maximum possible rate, upperbounded by the following ROS params -->
		<param name="update_airsim_img_response_every_n_sec" type="double" value="0.01" /> 
		<param name="update_airsim_control_every_n_sec" type="double" value="0.01" />
		<param name="update_lidar_every_n_sec" type="double" value="0.01" />
		<param name="publish_clock" type="bool" value="$(arg publish_clock)" />
		<param name="host_ip" type="string" value="$(arg host)" />
		<param name="world_frame_id" type="string" value="world_enu" />
		<param name="odom_frame_id" type="string" value="odom_local_enu" />
		<param name="coordinate_system_enu" type="bool" value="true" />
		
		<remap from="/airsim_node/Husky/FrontCamera/Scene" to="/oak_front/color/image_raw" />
		<remap from="/airsim_node/Husky/FrontCamera/Scene/camera_info" to="/oak_front/color/camera_info" />
		<remap from="/airsim_node/Husky/LeftCamera/Scene" to="/oak_left/color/image_raw" />
		<remap from="/airsim_node/Husky/LeftCamera/Scene/camera_info" to="/oak_left/color/camera_info" />
		<remap from="/airsim_node/Husky/RightCamera/Scene" to="/oak_right/color/image_raw" />
		<remap from="/airsim_node/Husky/RightCamera/Scene/camera_info" to="/oak_right/color/camera_info" />
	</node>

	<include file="$(find airsim_ros_pkgs)/launch/static_transforms.launch"/>

	<include file="$(find husky_control)/launch/teleop.launch"/>
	<node name="airsim_conversions" pkg="topological_mapping" type="ned2enu_airsim.py" />
</launch>
